### DISCUSSION TITLE
javascript import failing with record plugin

### DISCUSSION THREAD

**Discussion by EZWrighter** [katspaugh/wavesurfer.js#3332] [Q&A] [answered]
When attempting to import wavesurfer and record plugin into an html page as non-esm modules, javascript cannot find RecordPlugin to create it.

Using this import with the record plugin: <script src="https://unpkg.com/wavesurfer.js@7"></script>  which seems to work fine.

<script src="https://unpkg.com/wavesurfer.js@7/dist/plugins/record.min.js"></script> 
This module is obfuscated when compressed so no export of RecordPlugin

<script src="https://unpkg.com/wavesurfer.js@7/dist/plugins/record.js"></script> 
This file also seems to be compressed and minimized, another bug?

Attempting to call in javascript:
wavesurfer.registerPlugin(RecordPlugin.create());

Causes two errors:
Uncaught SyntaxError: Unexpected token 'export' (at record.js:1:4818)

and

Uncaught ReferenceError: RecordPlugin is not defined

Maybe I am not importing them properly?  But the compressed nature of record.js seem suspect. 

Thanks!

**Comment by EZWrighter** [comment]
Might just be a documentation thing on the page...this incantation seemed to do the trick: wavesurfer.registerPlugin(WaveSurfer.Record.create());

Using the minified js distribution:
https://unpkg.com/wavesurfer.js@7/dist/plugins/record.min.js

I still think there is a bug with the build of record.js since it's minified as well and errors out on load.

**Reply by katspaugh** [reply, marked as answer]
Yes, `WaveSurfer.Record` is correct for the min.js file.
This is mentioned in the readme:

<img width="876" alt="Screenshot 2023-11-10 at 10 47 37" src="https://github.com/katspaugh/wavesurfer.js/assets/381895/d3917361-43e8-4e41-8012-8f182a7c2d7c">

There's no bug in the .js file. It's minified but not called .min.js because it's not a UMD module but an ESM module (uses `export`).

**Reply by EZWrighter** [reply]
Thanks!  I feel like a knucklehead now :-D  I read that document section like 20 times.  Sorry for the bother.

**Comment by jagthedrummer** [comment]
I don't know if my project is setup weird or what but the `import` statements for plugins from the docs don't work for me.

When I use this import statement:

```javascript
import Regions from 'wavesurfer.js/dist/plugins/regions.js'
```

I get this error when trying to build the project:

```
Module not found: Error: Package path ./dist/plugins/regions is not exported from package
/private/var/folders/fx/vmqbtyws34j1l88m16_9jy_00000gn/T/broccoli-564051rEj26Bqm23F/cache-608-webpack_bundler_ember_auto_import_webpack/node_modules/wavesurfer.js
(see exports field in /private/var/folders/fx/vmqbtyws34j1l88m16_9jy_00000gn/T/broccoli-564051rEj26Bqm23F/cache-608-webpack_bundler_ember_auto_import_webpack/node_modules/wavesurfer.js/package.json)
```

If I remove `dist/` from the middle of the `import` path and also remove `.js` from the end then everything works well:

```javascript
import RegionsPlugin from 'wavesurfer.js/plugins/regions'
```

My project an Ember project and the error message indicates that `ember-auto-import` is involved in trying to resolve the plugin, so it might be an Ember thing and not a Wavesurfer thing. Just thought I'd mention it in case it helps anyone else (possibly including me in the future).
